database.cc

```diff
@@ -334,22 +334,25 @@ database::database(const db::config& cfg, database_config dbcfg, service::migrat
     , _read_concurrency_sem(max_count_concurrent_reads,
```

Comment:
Why is this a per-requisite? Looking at the next commit I see the max result size being stored in the permit itself. This makes sens: the limit is supplied per-read by the coordinator and it cannot be per-semaphore. So I don't understand why this is needed.